update to webpack 4

RoryO opened the pull request:
Couple of optimizations:

- Pull all of `node_modules` into a separate bundle. Without doing so the app takes several seconds to recompile every change
- Disabled the irritating and incorrect 'file size too large' warning from Webpack. The Phaser bundle is 6mb, that warning is irrelevant to this project.
- Added Babel and the standard preset. It isn't necessary though it's difficult to write with any new JS features without it.
- Alphabetize the webpack config

didaquis commented on package.json:
```diff
@@ -1,11 +1,11 @@
   "description": "A Phaser 3 Project Template",
   "main": "src/index.js",
   "scripts": {
     "build": "webpack",
```
I'm not an expert about Webpack, but if you apply this changes the code works. 
```
"build": "webpack --mode=development",
"start": "npm run build && webpack-dev-server --port=8000 --mode=development"
```

RoryO commented:
I'm pretty sure I cloned this fresn, ran `npm install`, `npm run` and it all went well. I will do it again when I have a moment.

didaquis commented:
hummm… maybe you have installed globally some module different than me? I’m not sure if that could be related

RoryO commented:
Cloning fresh, npm install, npm run

Windows:

```
PS A:\tmp\phaser3-project-template> node --version
v10.12.0

PS A:\tmp> git clone git@github.com:RoryO/phaser3-project-template.git
Cloning into 'phaser3-project-template'...
key_load_public: invalid format
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (5/5), done.
Receiving objects:  93% (70/75)    2 (delta 0), pack-reused 70 eceiving objects:  92% (69/75)
Receiving objects: 100% (75/75), 187.95 KiB | 6.71 MiB/s, done.
Resolving deltas: 100% (34/34), done.
PS A:\tmp> cd .\phaser3-project-template\
PS A:\tmp\phaser3-project-template> git checkout webpack-4
Switched to a new branch 'webpack-4'
Branch 'webpack-4' set up to track remote branch 'webpack-4' from 'origin'.
PS A:\tmp\phaser3-project-template> npm install
npm WARN deprecated socks@1.1.10: If using 2.x branch, please upgrade to at least 2.1.6 to avoid a serious bug with socket data flow and an import issue introduced in 2.1.0

> phaser@3.15.1 postinstall A:\tmp\phaser3-project-template\node_modules\phaser
> node scripts/support.js

❤ Please help support Phaser development ❤
https://www.patreon.com/photonstorm/

npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1029 packages from 1087 contributors and audited 14720 packages in 24.409s
found 0 vulnerabilities

PS A:\tmp\phaser3-project-template> npm run
Lifecycle scripts included in phaser3-project-template:
  start
    webpack-dev-server --port=8000 --mode=development

available via `npm run-script`:
  build
    webpack
PS A:\tmp\phaser3-project-template> npm run start

> phaser3-project-template@1.1.0 start A:\tmp\phaser3-project-template
> webpack-dev-server --port=8000 --mode=development

i ｢wds｣: Project is running at http://localhost:8000/
i ｢wds｣: webpack output is served from /build/
i ｢wdm｣: Hash: cd26e0d1dc6edbbf3e6d
Version: webpack 4.24.0
Time: 2908ms
Built at: 11/03/2018 4:40:37 PM
    Asset      Size  Chunks                    Chunk Names
 index.js  8.72 KiB   index  [emitted]         index
vendor.js  6.45 MiB  vendor  [emitted]  [big]  vendor
Entrypoint index [big] = vendor.js index.js
[0] multi (webpack)-dev-server/client?http://localhost:8000 ./src/index.js 40 bytes {index} [built]
[./node_modules/ansi-html/index.js] 4.16 KiB {vendor} [built]
[./node_modules/ansi-regex/index.js] 135 bytes {vendor} [built]
[./node_modules/events/events.js] 8.13 KiB {vendor} [built]
[./node_modules/html-entities/index.js] 231 bytes {vendor} [built]
[./node_modules/loglevel/lib/loglevel.js] 7.68 KiB {vendor} [built]
[./node_modules/phaser/src/phaser.js] 2.21 KiB {vendor} [built]
[./node_modules/strip-ansi/index.js] 161 bytes {vendor} [built]
[./node_modules/url/url.js] 22.8 KiB {vendor} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8000] (webpack)-dev-server/client?http://localhost:8000 7.78 KiB {vendor} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.58 KiB {vendor} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.05 KiB {vendor} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {index} [built]
[./node_modules/webpack/hot/emitter.js] (webpack)/hot/emitter.js 75 bytes {vendor} [built]
[./src/index.js] 488 bytes {index} [built]
    + 1157 hidden modules
i ｢wdm｣: Compiled successfully.
```

Linux

```
╭─rory@Stan /mnt/c/Users/rto
╰─$ node --version
v10.13.0
╭─rory@Stan /mnt/c/Users/rto
╰─$ git clone git@github.com:roryo/phaser3-project-template
Cloning into 'phaser3-project-template'...
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 75 (delta 0), reused 2 (delta 0), pack-reused 70
Receiving objects: 100% (75/75), 187.95 KiB | 206.00 KiB/s, done.
Resolving deltas: 100% (34/34), done.
╭─rory@Stan /mnt/c/Users/rto
╰─$ cd phaser3-project-template
╭─rory@Stan /mnt/c/Users/rto/phaser3-project-template  ‹master›
╰─$ git checkout webpack-4
Branch 'webpack-4' set up to track remote branch 'webpack-4' from 'origin'.
Switched to a new branch 'webpack-4'
╭─rory@Stan /mnt/c/Users/rto/phaser3-project-template  ‹webpack-4›
╰─$ npm install
npm WARN deprecated socks@1.1.10: If using 2.x branch, please upgrade to at least 2.1.6 to avoid a serious bug with socket data flow and an import issue introduced in 2.1.0
npm WARN lifecycle The node binary used for scripts is /home/rory/.asdf/shims/node but npm is using /home/rory/.asdf/installs/node/10.13.0/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> phaser@3.15.1 postinstall /mnt/c/Users/rto/phaser3-project-template/node_modules/phaser
> node scripts/support.js

❤ Please help support Phaser development ❤
https://www.patreon.com/photonstorm/

npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 1029 packages from 1087 contributors and audited 14720 packages in 134.516s
found 0 vulnerabilities



   ╭───────────────────────────────────────────────────────────────╮
   │                                                               │
   │       New minor version of npm available! 5.5.1 → 5.6.0       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v5.6.0   │
   │               Run npm install -g npm to update!               │
   │                                                               │
   ╰───────────────────────────────────────────────────────────────╯

╭─rory@Stan /mnt/c/Users/rto/phaser3-project-template  ‹webpack-4*›
╰─$ npm run start
npm WARN lifecycle The node binary used for scripts is /home/rory/.asdf/shims/node but npm is using /home/rory/.asdf/installs/node/10.13.0/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> phaser3-project-template@1.1.0 start /mnt/c/Users/rto/phaser3-project-template
> webpack-dev-server --port=8000 --mode=development

ℹ ｢wds｣: Project is running at http://localhost:8000/
ℹ ｢wds｣: webpack output is served from /build/
ℹ ｢wdm｣: Hash: cd26e0d1dc6edbbf3e6d
Version: webpack 4.24.0
Time: 3130ms
Built at: 11/03/2018 4:01:46 PM
    Asset      Size  Chunks                    Chunk Names
 index.js  8.72 KiB   index  [emitted]         index
vendor.js  6.45 MiB  vendor  [emitted]  [big]  vendor
Entrypoint index [big] = vendor.js index.js
[0] multi (webpack)-dev-server/client?http://localhost:8000 ./src/index.js 40 bytes {index} [built]
[./node_modules/ansi-html/index.js] 4.16 KiB {vendor} [built]
[./node_modules/ansi-regex/index.js] 135 bytes {vendor} [built]
[./node_modules/events/events.js] 8.13 KiB {vendor} [built]
[./node_modules/html-entities/index.js] 231 bytes {vendor} [built]
[./node_modules/loglevel/lib/loglevel.js] 7.68 KiB {vendor} [built]
[./node_modules/phaser/src/phaser.js] 2.21 KiB {vendor} [built]
[./node_modules/strip-ansi/index.js] 161 bytes {vendor} [built]
[./node_modules/url/url.js] 22.8 KiB {vendor} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8000] (webpack)-dev-server/client?http://localhost:8000 7.78 KiB {vendor} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.58 KiB {vendor} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.05 KiB {vendor} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {index} [built]
[./node_modules/webpack/hot/emitter.js] (webpack)/hot/emitter.js 75 bytes {vendor} [built]
[./src/index.js] 488 bytes {index} [built]
    + 1157 hidden modules
ℹ ｢wdm｣: Compiled successfully.
```

I don't know what the issue is. Can you clone it fresh as shown, install and run and see what happens?

didaquis commented:
I cloned your pull request on a new folder (fresh cloning), after that I run command `npm install` and then `npm run build`. 

```
> phaser3-project-template@1.1.0 build /Users/didac/repos_ajenos/temporal
> webpack

Hash: f0e0313d35a8dc5e0789
Version: webpack 4.24.0
Time: 530ms
Built at: 08/11/2018 20:46:51
 1 asset
Entrypoint index = index.js
[0] ./src/index.js 3 KiB {0} [built] [failed] [1 error]

WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/

ERROR in ./src/index.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
Error: Cannot find module 'babel-preset-env' from '/Users/didac/repos_ajenos/temporal'
- Did you mean "@babel/env"?
    at Function.module.exports [as sync] (/Users/didac/repos_ajenos/temporal/node_modules/resolve/lib/sync.js:43:15)
    at resolveStandardizedName (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/files/plugins.js:101:31)
    at resolvePreset (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/files/plugins.js:58:10)
    at loadPreset (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/files/plugins.js:77:20)
    at createDescriptor (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-descriptors.js:154:9)
    at items.map (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-descriptors.js:109:50)
    at Array.map (<anonymous>)
    at createDescriptors (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-descriptors.js:109:29)
    at createPresetDescriptors (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-descriptors.js:101:10)
    at passPerPreset (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-descriptors.js:58:96)
    at cachedFunction (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/caching.js:33:19)
    at presets.presets (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-descriptors.js:29:84)
    at mergeChainOpts (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-chain.js:315:26)
    at /Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-chain.js:278:7
    at buildRootChain (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/config-chain.js:68:29)
    at loadPrivatePartialConfig (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/partial.js:85:55)
    at Object.loadPartialConfig (/Users/didac/repos_ajenos/temporal/node_modules/@babel/core/lib/config/partial.js:110:18)
    at Object.<anonymous> (/Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:140:26)
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:3:103)
    at _next (/Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:5:194)
    at /Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:5:364
    at new Promise (<anonymous>)
    at Object.<anonymous> (/Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:5:97)
    at Object._loader (/Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:220:18)
    at Object.loader (/Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:56:18)
    at Object.<anonymous> (/Users/didac/repos_ajenos/temporal/node_modules/babel-loader/lib/index.js:51:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! phaser3-project-template@1.1.0 build: `webpack`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the phaser3-project-template@1.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

photonstorm commented:
Thank you for submitting this PR, I'm going to close it as we've just updated the repo to webpack 4.
